### ISSUE TITLE
showLicensePage throws RenderFlex overflow error in landscape when fontSize/Display size is largest 

### ISSUE THREAD

**Issue by arsenioo** [flutter/flutter#65177]
## Steps to Reproduce

1. Create test app: flutter create test.
2. Change fab onPressed method to be like this:
```
onPressed: () => showLicensePage(
	context: context,
	applicationIcon: Icon(Icons.add),
	applicationVersion: "TEST",
),
```
3. Start x86 Emulator Pixel 2 1980x1920 xxhdpi
4. Open Settings/Display/Advanced and set Font Size to Larger and Display Size to Largest
5. Rotate emulator in landscape mode
6. Execute **flutter run** to start app
7. Click on fab to show license page

**Expected results:** 
Licenses list without any issues

**Actual results:** 
Black/Yellow markups showing overflow error at the bottom of the screen until licenses list is rendered completely. The error present only during progress indicator stage, then disappears.

<details>
  <summary>Logs</summary>

```
I/flutter (16356): ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (16356): The following assertion was thrown during layout:
I/flutter (16356): A RenderFlex overflowed by 1.00 pixels on the bottom.
I/flutter (16356): 
I/flutter (16356): The relevant error-causing widget was:
I/flutter (16356):   MaterialApp file:///Users/apc/Work/test2/lib/main.dart:12:12
I/flutter (16356): 
I/flutter (16356): The overflowing RenderFlex has an orientation of Axis.vertical.
I/flutter (16356): The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and
I/flutter (16356): black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
I/flutter (16356): Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the
I/flutter (16356): RenderFlex to fit within the available space instead of being sized to their natural size.
I/flutter (16356): This is considered an error condition because it indicates that there is content that cannot be
I/flutter (16356): seen. If the content is legitimately bigger than the available space, consider clipping it with a
I/flutter (16356): ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex,
I/flutter (16356): like a ListView.
I/flutter (16356): The specific RenderFlex in question is: RenderFlex#00445 relayoutBoundary=up4 OVERFLOWING:
I/flutter (16356):   creator: Column ← LayoutBuilder-[<ConnectionState.waiting>] ← FadeTransition ← KeyedSubtree-[<0>] ←
I/flutter (16356):     Stack ← AnimatedSwitcher ← FutureBuilder<_LicenseData> ← _PackagesView ← _BodyBuilder ← MediaQuery
I/flutter (16356):     ← LayoutId-[<_ScaffoldSlot.body>] ← CustomMultiChildLayout ← ⋯
I/flutter (16356):   parentData: offset=Offset(0.0, 0.0) (can use size)
I/flutter (16356):   constraints: BoxConstraints(0.0<=w<=520.9, 0.0<=h<=240.0)
I/flutter (16356):   size: Size(520.9, 240.0)
I/flutter (16356):   direction: vertical
I/flutter (16356):   mainAxisAlignment: start
I/flutter (16356):   mainAxisSize: max
I/flutter (16356):   crossAxisAlignment: center
I/flutter (16356):   verticalDirection: down
I/flutter (16356): ◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤
I/flutter (16356): ════════════════════════════════════════════════════════════════════════════════════════════════════

```

```
[✓] Flutter (Channel master, 1.22.0-10.0.pre.65, on Mac OS X 10.15.6 19G2021, locale en-UA)
    • Flutter version 1.22.0-10.0.pre.65 at /Users/apc/Library/flutter
    • Framework revision 2032a448ed (7 hours ago), 2020-09-02 23:36:19 -0700
    • Engine revision 165abef0a2
    • Dart version 2.10.0 (build 2.10.0-77.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/apc/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/apc/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/apc/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the
        Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install:
        sudo gem install cocoapods

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (1 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86 • Android 11 (API 30) (emulator)

! Doctor found issues in 1 category.
```

</details>

**Comment by arsenioo** [comment]
@TonicArtos

**Comment by TahaTesser** [comment]
Hi @arsenioo 
I can reproduce the issue 

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: Text('Hello World'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () => showLicensePage(
          context: context,
          applicationIcon: Icon(Icons.add),
          applicationVersion: "TEST",
        ),
      ),
    );
  }
}


```

</details> 

<details> 
<summary>logs</summary> 

```
Launching lib\main.dart on Android SDK built for x86 in debug mode...
√ Built build\app\outputs\flutter-apk\app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:60058/--Z1ryBcstU=/ws
D/EGL_emulation( 8330): eglMakeCurrent: 0xe8da9f00: ver 2 0 (tinfo 0xdd9f2c90)
D/eglCodecCommon( 8330): setVertexArrayObject: set vao to 0 (0) 1 0
D/EGL_emulation( 8330): eglMakeCurrent: 0xf407f700: ver 2 0 (tinfo 0xdd9f2a80)
D/EGL_emulation( 8330): eglMakeCurrent: 0xe8da9f00: ver 2 0 (tinfo 0xdd9f2c90)
D/EGL_emulation( 8330): eglMakeCurrent: 0xf407f700: ver 2 0 (tinfo 0xdd9f2a80)

[38;5;248m════════ Exception caught by rendering library ═════════════════════════════════[39;49m
[38;5;244mThe following assertion was thrown during layout:[39;49m
A RenderFlex overflowed by 1.00 pixels on the bottom.

[38;5;244mThe relevant error-causing widget was[39;49m
    [38;5;248mMaterialApp[39;49m
[38;5;244mThe overflowing RenderFlex has an orientation of Axis.vertical.[39;49m
[38;5;244mThe edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and black striped pattern. This is usually caused by the contents being too big for the RenderFlex.[39;49m

[38;5;248mConsider applying a flex factor (e.g. using an Expanded widget) to force the children of the RenderFlex to fit within the available space instead of being sized to their natural size.[39;49m
[38;5;248mThis is considered an error condition because it indicates that there is content that cannot be seen. If the content is legitimately bigger than the available space, consider clipping it with a ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex, like a ListView.[39;49m

[38;5;244mThe specific RenderFlex in question is: RenderFlex#3cb7a relayoutBoundary=up4 OVERFLOWING[39;49m
[38;5;248m════════════════════════════════════════════════════════════════════════════════[39;49m

[38;5;248m════════ Exception caught by rendering library ═════════════════════════════════[39;49m
A RenderFlex overflowed by 1.00 pixels on the bottom.
[38;5;244mThe relevant error-causing widget was[39;49m
    [38;5;248mMaterialApp[39;49m
[38;5;248m════════════════════════════════════════════════════════════════════════════════[39;49m

[38;5;248m════════ Exception caught by rendering library ═════════════════════════════════[39;49m
A RenderFlex overflowed by 1.00 pixels on the bottom.
[38;5;244mThe relevant error-causing widget was[39;49m
    [38;5;248mMaterialApp[39;49m
[38;5;248m════════════════════════════════════════════════════════════════════════════════[39;49m

``` 

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-10.0.pre.65, on Microsoft Windows [Version 10.0.19041.450], locale en-US)
    • Flutter version 1.22.0-10.0.pre.65 at C:\Code\flutter_master
    • Framework revision 2032a448ed (9 hours ago), 2020-09-02 23:36:19 -0700
    • Engine revision 165abef0a2
    • Dart version 2.10.0 (build 2.10.0-77.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30413.136
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.48.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.13.2

[✓] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.450]
    • Web Server (web)                   • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 84.0.4147.135

• No issues found!
```

</details>

**Comment by HansMuller** [comment]
CC @gspencergoog

**Comment by gspencergoog** [comment]
@TonicArtos are you able to take a look at this?

**Comment by TonicArtos** [comment]
> @TonicArtos are you able to take a look at this?

No problems. I'll get it done.

**Comment by exaby73** [comment]
I cannot reproduce this issue with `Master (3.6.0-3.0.pre.54)`. Closing this issue as fixed. If you disagree, please comment with a [complete and minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) and I can reopen it. Thank you.
